src/GraphQL/Execution/ExecutionContext.cs

```diff
@@ -62,20 +63,27 @@ public class ExecutionContext : IExecutionContext
 
         private readonly List<Array> _trackedArrays = new List<Array>();
 
-        internal void TrackArray(Array array)
-        {
-            if (array == null)
-                throw new ArgumentNullException(nameof(array));
+        TElement[] IExecutionArrayPool.Rent<TElement>(int minimumLength) => RentSharedArray<TElement>(minimumLength);
 
+        /// <inheritdoc cref="IExecutionArrayPool.Rent{TElement}(int)"/>
+        public TElement[] RentSharedArray<TElement>(int minimumLength)
+        {
+            var array = ArrayPool<TElement>.Shared.Rent(minimumLength);
             lock (_trackedArrays)
                 _trackedArrays.Add(array);
+            return array;
         }
 
-        internal void ReturnArrays()
+        /// <summary>
+        /// Releases any rented arrays back to the backing memory pool.
+        /// </summary>
+        public void Dispose()
```

Comment:
I deliberately didn't inherit `ExecutionContext` from `IDisposable`. Do you think it's worth doing this?

Reply:
OK. I think indeed it is worth.